packages/tools/src/lib/build-cache/remoteBuildCache.ts

```diff
@@ -1,15 +1,18 @@
 import type { RemoteBuildCache } from './common.js';
 
 export async function createRemoteBuildCache(
-  remoteCacheProvider: 'github-actions' | null | { new (): RemoteBuildCache }
+  remoteCacheProvider: 'github-actions' | null | (() => RemoteBuildCache)
 ): Promise<RemoteBuildCache | null> {
   if (remoteCacheProvider === 'github-actions') {
-    const { GitHubBuildCache } = await import('./github/GitHubBuildCache.js');
-    return new GitHubBuildCache();
+    const { pluginGitHubBuildCache } = await import(
+      './github/GitHubBuildCache.js'
+    );
+    const gitHubCacheProvider = pluginGitHubBuildCache();
+    return gitHubCacheProvider();
```

Comment:
could you please explain what's the point of plugins here? why we cannot import class and initialize it with relevant properties.

Reply:
I didn't want our internal provider to be special, that's why this is a bit more convoluted than it needs to be. I expect devs will copy this for other providers and this will allow them to pass initial params in a correct way